### ISSUE TITLE
Remove reference to UNSAFE_componentWillUpdate

### ISSUE THREAD

**Issue by TheRealNate** [TheRealNate/meteor-react-native#3]
`UNSAFE_componentWillUpdate` is a legacy method

/src/components/ReactMeteorData.js#L19

**Comment by PolarisWT** [comment]
This update is a bit more complex than #4.  Have you thought about how to best handle this already?

For the short-term, could we just use the `UNSAFE_componentWillUpdate()` which is [scheduled](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#gradual-migration-path) to last through 17.0+?  This should suppress the warning.

Longer term, looking at `componentDidUpdate()` may work with corrections for arguments that are backwards looking (`prevProps`, `prevState`) vs the current forward looking (`nextProps`, `nextState`)

**Comment by TheRealNate** [comment]
@PolarisWT I have merged your PR for renaming to UNSAFE_. 

Definitely a tricky update. The author states:
````
Temporarily assign this.state and this.props, so that they are seen by getMeteorData!

This is a simulation of how the proposed Observe API for React will work, which calls observe() after componentWillUpdate and after props and state are updated, but before render() is called. See https://github.com/facebook/react/issues/3398.
```` 

So it seems that `this._meteorDataManager.calculateData` needs to be run with new props and state but before render. `componentDidUpdate` is run *after* render.

The [suspense](https://reactjs.org/docs/concurrent-mode-suspense.html) API may be a solution (it's mentioned in the "Observe" proposal"), but technically isn't ready for stable release yet...

**Comment by TheRealNate** [comment]
This has been fixed in `2.1.0`
